server/integration/testsuite/src/test/java/org/infinispan/server/test/client/hotrod/AbstractRemoteCacheIT.java

```diff
@@ -78,17 +86,54 @@ public static void release() {
         }
     }
 
-    private Configuration createRemoteCacheManagerConfiguration() {
+    protected static Archive<?> createFilterArchive() {
+        return ShrinkWrap.create(JavaArchive.class, "filter.jar")
+                .addClasses(StaticCacheEventFilterFactory.class, DynamicCacheEventFilterFactory.class,
+                        CustomPojoEventFilterFactory.class, Person.class)
+                .addAsServiceProvider(CacheEventFilterFactory.class,
+                        StaticCacheEventFilterFactory.class, DynamicCacheEventFilterFactory.class,
+                        CustomPojoEventFilterFactory.class);
+    }
+
+    protected static Archive<?> createConverterArchive() {
+        return ShrinkWrap.create(JavaArchive.class, "converter.jar")
+                .addClasses(StaticCacheEventConverterFactory.class, DynamicCacheEventConverterFactory.class,
+                        CustomPojoEventConverterFactory.class, Person.class, CustomEvent.class)
+                .addAsServiceProvider(CacheEventConverterFactory.class,
+                        StaticCacheEventConverterFactory.class, DynamicCacheEventConverterFactory.class,
+                        CustomPojoEventConverterFactory.class);
+    }
+
+    protected static Archive<?> createFilterConverterArchive() {
+        return ShrinkWrap.create(JavaArchive.class, "filter-converter.jar")
+                .addClasses(FilterConverterFactory.class, CustomEvent.class,
+                        CustomPojoFilterConverterFactory.class, Person.class, Id.class)
+                .addAsServiceProvider(CacheEventFilterConverterFactory.class, FilterConverterFactory.class,
+                        CustomPojoFilterConverterFactory.class);
+    }
+
+    protected static Archive<?> createKeyValueFilterConverterArchive() {
+        return ShrinkWrap.create(JavaArchive.class, "key-value-filter-converter.jar")
+                .addClasses(TestKeyValueFilterConverterFactory.class, SampleEntity.class, Summary.class, SampleEntity.SampleEntityExternalizer.class, Summary.SummaryExternalizer.class)
+                .addAsServiceProvider(KeyValueFilterConverterFactory.class, TestKeyValueFilterConverterFactory.class);
+    }
+
+    private Configuration createRemoteCacheManagerConfiguration(int... hotrodPortOverrides) {
         ConfigurationBuilder config = new ConfigurationBuilder();
+        if (hotrodPortOverrides.length != 0 && getServers().size() != hotrodPortOverrides.length) {
+            throw new IllegalArgumentException("The number of defined ports is different from server count");
```

Comment:
How about using assert here? Assertions fits better into tests and this part sounds like a preconditions check.